### PR TITLE
[SMD-491] - terraform support filters

### PR DESCRIPTION
**Pull request by adussarps**
This PR adds supports for suppress and require filters for the rules ressources.

### CODE REVIEW

File: datadog/data_source_datadog_security_monitoring_rules.go [hunk 1]
```diff
@@ -228,6 +228,11 @@ func buildSecurityMonitoringTfRule(rule datadogV2.SecurityMonitoringRuleResponse
 		tfRule["tags"] = *tags
 	}
 
+	if _, ok:= rule.GetFiltersOk(); ok{
+		filters := extractFiltersFromRuleResponse(rule)
```
Comment (mellon85) [inline datadog/data_source_datadog_security_monitoring_rules.go:232]: why not just

```
tfRule["filter"] = extractFiltersFromRuleResponse(rule)
```

**Review by mellon85 — APPROVED** [review]

File: datadog/resource_datadog_security_monitoring_rule.go [hunk 2]
```diff
@@ -432,6 +477,7 @@ func updateResourceDataFromResponse(d *schema.ResourceData, ruleResponse datadog
 	d.Set("case", ruleCases)
 	d.Set("message", ruleResponse.GetMessage())
 	d.Set("name", ruleResponse.GetName())
+	d.Set("has_extended_title", ruleResponse.GetHasExtendedTitle())
```
Comment (greentruff) [inline datadog/resource_datadog_security_monitoring_rule.go:480]: Is there a test which checks this change?

Comment (adussarps) [reply datadog/resource_datadog_security_monitoring_rule.go:480]: well it should be in the previous PR actually 👀

File: datadog/data_source_datadog_security_monitoring_rules.go (hunk 1 above)
Comment (adussarps) [reply datadog/data_source_datadog_security_monitoring_rules.go:232]: not needed indeed 👍

File: datadog/resource_datadog_security_monitoring_rule.go (hunk 2 above)
Comment (zippolyte) [reply datadog/resource_datadog_security_monitoring_rule.go:478]: I'm creating a separate PR to fix this, let's remove from this one
```suggestion
```

File: datadog/resource_datadog_security_monitoring_rule.go [hunk 3]
```diff
@@ -387,6 +414,24 @@ func buildCreatePayloadQueries(d *schema.ResourceData) []datadogV2.SecurityMonit
...
+func buildPayloadFilters(tfFilters []interface{}) []datadogV2.SecurityMonitoringFilter {
+	payloadFilters := make([]datadogV2.SecurityMonitoringFilter, len(tfFilters))
+	for idx, tfFilter := range tfFilters {
+		filter := tfFilter.(map[string]interface{})
+		payloadFilter := datadogV2.SecurityMonitoringFilter{}
+
+		if v, ok := filter["action"]; ok {
+			action := datadogV2.SecurityMonitoringFilterAction(v.(string))
+			payloadFilter.SetAction(action)
+		}
```
[the remark this answers is not in the document]
Comment (zippolyte) [reply datadog/resource_datadog_security_monitoring_rule.go:426]: Unless it should be marked optional in the schema I don't know

**Review by zippolyte — CHANGES_REQUESTED** [review]

File: datadog/resource_datadog_security_monitoring_rule.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (adussarps) [reply datadog/resource_datadog_security_monitoring_rule.go]: Removed it and updated the cassettes 👍

**Comment by zippolyte** [comment]
I think you can remove your last commit updating cassettes. It's probably not needed. Also you should rebase on the latest master :)

**Review by zippolyte — APPROVED** [review]

**Pull request merged**